### PR TITLE
get-resources.only run backup logic if resource is tutorial

### LINKED ISSUE #4636: Spurious backup operation on restart
- `cylc play myflow` and stop it before it finishes
```
INFO - Extracting job.sh to ~/cylc-run/myflow/.service/etc/job.sh
...
```

So far so good. Now:
- `cylc play myflow` again to restart
```diff
 INFO - Extracting job.sh to ~/cylc-run/myflow/.service/etc/job.sh
+WARNING - Replacing an existing cylc-tutorials folder which will be copied to ~/cylc-run/myflow/.service/etc/job.sh.20220131T175412Z
...
```

Assigning @wxtim as this looks to be related to the get-resources change

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by wxtim**
These changes close #4636 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry needed (follow up to a 8.0rc1 PR).
- [x] No documentation update required.

### CODE REVIEW

**Comment by wxtim** [comment]
@MetRonnie  - Might I ask if you can test the fix empirically, then I'll test it formally?

File: cylc/flow/resources.py [hunk 1]
```diff
@@ -128,13 +128,13 @@ def _backup(tgt: Path) -> None:
...
 
-def extract_resource(src: Path, tgt: Path) -> None:
+def extract_resource(src: Path, tgt: Path, is_tutorial: bool = False) -> None:
     """Extract src into tgt.
 
     NOTE: src can be a dir or a file.
     """
     LOG.info(f"Extracting {src.relative_to(RESOURCE_DIR)} to {tgt}")
-    if tgt.exists():
+    if tgt.exists() and is_tutorial:
```
Comment (MetRonnie) [inline cylc/flow/resources.py:137]: Probably better to skip this filesystem operation unless needed:
```suggestion
    if is_tutorial and tgt.exists():
```

**Review by MetRonnie — COMMENTED** [review]
Tested, works

**Comment by oliver-sanders** [comment]
Alternatively make the backup optional and only enable it when used via the `cylc get-resources` CLI.

**Review by MetRonnie — APPROVED** [review]

**Pull request merged**